language_server/src/logger.h

```diff
@@ -15,44 +15,57 @@
 #ifndef HLASMPLUGIN_HLASMLANGUAGESERVER_LOGGER_H
 #define HLASMPLUGIN_HLASMLANGUAGESERVER_LOGGER_H
 
+#include <concepts>
 #include <mutex>
+#include <span>
 #include <string_view>
 
 namespace hlasm_plugin::language_server {
 
-// LOG_INFO is disabled in release, because it writes vast amount of text
-// which takes significant time
-#ifdef _DEBUG
-#    define LOG_ON
-#endif
+#define LOG_WRITE(l, ...)                                                                                              \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        const unsigned level = l;                                                                                      \
+        if (level > hlasm_plugin::language_server::logger::instance.level())                                           \
+            break;                                                                                                     \
+        hlasm_plugin::language_server::logger::instance.log(level, __VA_ARGS__);                                       \
+    } while (false)
```

Comment:
Why does this need to be in a `do` - `while` loop?

Reply:
To force the `;` after the statement safely.